src/Editor/ImGuizmo/imguizmo.cpp

```diff
@@ -0,0 +1,1719 @@
+            for (int j = 0; j < stepCount; j++)
+            {
+                float t1 = (float)j * stepLength;
+                float t2 = (float)j * stepLength + stepLength * 0.5f;
+                ImVec2 worldBoundSS1 = ImLerp(worldBound1, worldBound2, ImVec2(t1, t1));
+                ImVec2 worldBoundSS2 = ImLerp(worldBound1, worldBound2, ImVec2(t2, t2));
+                drawList->AddLine(worldBoundSS1, worldBoundSS2, 0xAAAAAA + anchorAlpha, 3.f);
+            }
+            vec_t midPoint = (aabb[i] + aabb[(i + 1) % 4]) * 0.5f;
+            ImVec2 midBound = worldToPos(midPoint, boundsMVP);
+            static const float AnchorBigRadius = 8.f;
+            static const float AnchorSmallRadius = 6.f;
+            bool overBigAnchor = ImLengthSqr(worldBound1 - io.MousePos) <= (AnchorBigRadius*AnchorBigRadius);
+            bool overSmallAnchor = ImLengthSqr(midBound - io.MousePos) <= (AnchorBigRadius*AnchorBigRadius);
+
+
+            unsigned int bigAnchorColor = overBigAnchor ? selectionColor : (0xAAAAAA + anchorAlpha);
+            unsigned int smallAnchorColor = overSmallAnchor ? selectionColor : (0xAAAAAA + anchorAlpha);
+
+            drawList->AddCircleFilled(worldBound1, AnchorBigRadius, bigAnchorColor);
+            drawList->AddCircleFilled(midBound, AnchorSmallRadius, smallAnchorColor);
+            int oppositeIndex = (i + 2) % 4;
+            // big anchor on corners
+            if (!gContext.mbUsingBounds && gContext.mbEnable && overBigAnchor && io.MouseDown[0])
+            {
+                gContext.mBoundsPivot.TransformPoint(aabb[(i + 2) % 4], gContext.mModelSource);
+                gContext.mBoundsAnchor.TransformPoint(aabb[i], gContext.mModelSource);
+                gContext.mBoundsPlan = BuildPlan(gContext.mBoundsAnchor, bestAxisWorldDirection);
+                gContext.mBoundsBestAxis = bestAxis;
+                gContext.mBoundsAxis[0] = secondAxis;
+                gContext.mBoundsAxis[1] = thirdAxis;
+
+                gContext.mBoundsLocalPivot.Set(0.f);
+                gContext.mBoundsLocalPivot[secondAxis] = aabb[oppositeIndex][secondAxis];
+                gContext.mBoundsLocalPivot[thirdAxis] = aabb[oppositeIndex][thirdAxis];
+
+                gContext.mbUsingBounds = true;
+                gContext.mBoundsMatrix = gContext.mModelSource;
+            }
+            // small anchor on middle of segment
+            if (!gContext.mbUsingBounds && gContext.mbEnable && overSmallAnchor && io.MouseDown[0])
+            {
+                vec_t midPointOpposite = (aabb[(i + 2) % 4] + aabb[(i + 3) % 4]) * 0.5f;
+                gContext.mBoundsPivot.TransformPoint(midPointOpposite, gContext.mModelSource);
+                gContext.mBoundsAnchor.TransformPoint(midPoint, gContext.mModelSource);
+                gContext.mBoundsPlan = BuildPlan(gContext.mBoundsAnchor, bestAxisWorldDirection);
+                gContext.mBoundsBestAxis = bestAxis;
+                int indices[] = { secondAxis , thirdAxis };
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') comma should not be preceded by whitespace [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=vera%2B%2B%3AT009)